tests/integration/run_integration_test.sh

```
@@ -0,0 +1,426 @@
+    if [[ -n "${HTTP_PID:-}" ]] && kill -0 "${HTTP_PID}" 2>/dev/null; then
+        kill "${HTTP_PID}" 2>/dev/null || true
+        wait "${HTTP_PID}" 2>/dev/null || true
+    fi
+
+    # Remove temp directories
+    if [[ -n "${WORK_DIR:-}" && -d "${WORK_DIR}" ]]; then
+        rm -rf "${WORK_DIR}"
+    fi
+
+    log_info "Cleanup complete."
+}
```

Shellcheck (0.11.0)

[info] This function is never invoked. Check usage (or ignored if invoked indirectly).

(SC2329)